src/cluster_linearize.h

```diff
@@ -184,6 +184,48 @@ class DepGraph
         }
     }
 
+    /** Compute the (reduced) set of parents of node i in this graph.
+     *
+     * This returns the minimal subset of the parents of i whose ancestors together equal all of
+     * i's ancestors (unless i is part of a cycle of dependencies). Note that DepGraph does not
+     * store the actual set of parents.
```

Comment:
nit: can go further with the explanation the comments with something like

```suggestion
     * i's ancestors (unless i is part of a cycle of dependencies). Note that DepGraph does not
     * store set of parents; it has to be reduced from ancestor set.
```

for both the parent and child reducing methods.